plugin/inlay_hint.py

```diff
@@ -0,0 +1,127 @@
+from .core.sessions import Session
+from .core.protocol import InlayHintLabelPart, MarkupContent, Point, InlayHint, Request
+from .core.registry import LspTextCommand
+from .core.typing import List, Optional, Union
+from .core.views import FORMAT_MARKUP_CONTENT, point_to_offset, minihtml
+from .formatting import apply_text_edits_to_view
+import html
+import sublime
+import uuid
+from .core.types import TemporarySettings
+
+
+class LspToggleInlayHintsCommand(LspTextCommand):
+    capability = 'inlayHintProvider'
+
+    def run(self, _edit: sublime.Edit, _event: Optional[dict] = None) -> None:
+        sessions = self.sessions('inlayHintProvider')
+        TemporarySettings.SHOW_INLAY_HINTS = not TemporarySettings.SHOW_INLAY_HINTS
+        for session in sessions:
+            for sv in session.session_views_async():
+                sv.session_buffer.do_inlay_hints_async(sv.view)
+
+
+class LspInlayHintApplyTextEditsCommand(LspTextCommand):
+    capability = 'inlayHintProvider'
+
+    def run(self, _edit: sublime.Edit, session_name: str, inlay_hint: InlayHint, phantom_uuid: str,
+            event: Optional[dict] = None) -> None:
+        session = self.session_by_name(session_name, 'inlayHintProvider')
+        if session and session.has_capability('inlayHintProvider.resolveProvider'):
+            request = Request.resolveInlayHint(inlay_hint, self.view)
+            session.send_request_async(request, lambda response: self.handle(session_name, response, phantom_uuid))
+            return
+        self.handle(session_name, inlay_hint, phantom_uuid)
+
+    def handle(self, session_name: str, inlay_hint: InlayHint, phantom_uuid: str) -> None:
+        self.handle_inlay_hint_text_edits(session_name, inlay_hint, phantom_uuid)
+
+    def handle_inlay_hint_text_edits(self, session_name: str, inlay_hint: InlayHint, phantom_uuid: str) -> None:
+        session = self.session_by_name(session_name, 'inlayHintProvider')
+        if not session:
+            return
+        text_edits = inlay_hint.get('textEdits')
+        if not text_edits:
+            return
+        for sv in session.session_views_async():
+            sv.remove_inlay_hint_phantom(phantom_uuid)
+        apply_text_edits_to_view(text_edits, self.view)
+
+
+def get_inlay_hint_html(view: sublime.View, inlay_hint: InlayHint, session: Session, phantom_uuid: str) -> str:
+    tooltip = format_inlay_hint_tooltip(view, inlay_hint.get("tooltip"))
+    label = format_inlay_hint_label(view, inlay_hint["label"])
+    margin_left = "0.6rem" if inlay_hint.get("paddingLeft", False) else "0"
+    margin_right = "0.6rem" if inlay_hint.get("paddingRight", False) else "0"
+    font = view.settings().get('font_face') or "monospace"
```

Comment:
Is it necessary to explicitly set font? Don't phantoms inherit view/ST font by default?

Reply:
Nope, On a Mac that I have, not setting the font-family will case the phantoms to have a sans-serif font